jdaviz/app.py

```diff
@@ -902,6 +903,9 @@ def add_data(self, data, data_label=None, notify_done=True):
         if not data_label and hasattr(data, "label"):
             data_label = data.label
         data_label = self.return_unique_name(data_label)
+        if data_label in self.data_collection.labels:
+            warnings.warn(f"Overwriting existing data entry with label '{data_label}'")
+
```

Comment:
related: #1709

Reply:
Is this even possible since this is returned by `data_label = self.return_unique_name(data_label)` above?

Reply:
Yes, this is needed to mimic the previous behavior in the `test_case_that_breaks_return_label` [test introduced in #1672](https://github.com/spacetelescope/jdaviz/pull/1672/files#diff-7c965473477cf991677e0a2a558724cf145d2b2765dde7fc77fe6d354e370915R137-R140)